sys/include/shell.h

```diff
@@ -202,6 +202,19 @@ static inline void shell_run(const shell_command_t *commands,
  *
  * This macro is a helper for defining a shell command and adding it to the
  * shell commands XFA (cross file array).
+ *
+ * @experimental This should be considered experimental API, subject to change
+ *               without notice!
```

Comment:
The API seems simple enough, are there really changes expected to it? I could imagine that something might change in the XFA backend, but this shouldn't effect this macro as far as I can judge.

Reply:
True, but as XFA is experimental, I'd rather not have external users depend on this for now. Time will tell if issues pop up with XFA in general...

Also, there might be a way to deduce the function name from the command name (e.g., by requiring the function to be called `_shell_command_<name>`). I'd like to be able to do that without going through API change procedures.

Reply:
> True, but as XFA is experimental, I'd rather not have external users depend on this for now. Time will tell if issues pop up with XFA in general...

This can be experimental without stating that the API is subject to change ;-).



> Also, there might be a way to deduce the function name from the command name (e.g., by requiring the function to be called `_shell_command_<name>`). I'd like to be able to do that without going through API change procedures.

I would rather make this a convenience function in addition to `SHELL_COMMAND()`. There is no technical reason for the user to not keep the freedom to call the handler whatever they want it to be called.

Reply:
> There is no technical reason for the user to not keep the freedom to call the handler whatever they want it to be called.

It would simplify searching the source code significantly, if a simple regex could detect any shell command implementation.

Reply:
>  there might be a way to deduce the function name from the command name (e.g., by requiring the function to be called `_shell_command_<name>`

I think this is one of those instances where the convenience of the auto-magic does not justify the added complexity of making it non-obvious what's going on.

Reply:
> This can be experimental without stating that the API is subject to change ;-).

It is our standard disclaimer for new APIs...

Reply:
Ok, I did not know that we have a standard disclaimer.